src/TestRunnerStartTask.ts

```diff
@@ -0,0 +1,273 @@
+
+const Query = require('@microfocus/alm-octane-js-rest-sdk/lib/query');
+
+export class TestRunnerStartTask extends BaseTask {
+
+    private executorId: string;
+    private executorName: string;
+    private executorSubtype: string;
+    private ciJobId: string;
+    private ciServer: any;
+    private executor: any;
+    private framework: string;
+    private testToConvert: string;
+
+    private constructor(tl: any) {
+        super(tl);
+    }
+
+    public static async instance(tl: any): Promise<TestRunnerStartTask> {
+        const task = new TestRunnerStartTask(tl);
+        await task.init(BaseTask.ALM_OCTANE_TEST_RUNNER_START);
+        return task;
+    }
+
+    protected prepareAzureVariables() {
+        this.collectionUri = this.tl.getVariable('System.TeamFoundationCollectionUri');
+        this.projectId = this.tl.getVariable('System.TeamProjectId');
+        this.projectName = this.tl.getVariable('System.TeamProject');
+        this.buildDefinitionName = this.tl.getVariable('Build.DefinitionName');
+        this.buildId = this.tl.getVariable('Build.BuildId');
+        this.sourceBranchName = this.tl.getVariable('Build.SourceBranchName');
+        this.jobStatus = this.convertJobStatus(this.tl.getVariable('AGENT_JOBSTATUS'));
+        this.definitionId = this.tl.getVariable("System.DefinitionId");
+        this.sourceBranch = this.tl.getVariable("Build.SourceBranch");
+
+        this.logger.info('collectionUri = ' + this.collectionUri);
+        this.logger.info('projectId = ' + this.projectId);
+        this.logger.info('projectName = ' + this.projectName);
+        this.logger.info('buildDefinitionName = ' + this.buildDefinitionName);
+        this.logger.info('sourceBranchName = ' + this.sourceBranchName);
+        this.logger.info('definitionId = ' + this.definitionId);
+        this.logger.info('sourceBranch = ' + this.sourceBranch);
+
+        this.initTestRunnerVariables();
+    }
+
+    private async initTestRunnerVariables() {
+        await new Promise<void>(async (resolve, reject) => {
+            try {
+                this.prepareFramework();
+                this.prepareTestToConvert();
+                resolve();
+            } catch (ex) {
+                reject(ex);
+            }
+        }).catch(ex => {
+            this.logger.error(ex);
+            this.tl.setResult(this.tl.TaskResult.Failed, 'Tests Converter initialization failed');
+
+            throw ex;
+        });
+    }
+
+    private prepareFramework():void {
+        this.framework = this.tl.getInput(InputConstants.FRAMEWORK, true);
+    }
+    private prepareTestToConvert(): void {
+        this.testToConvert = this.tl.getVariable('testsToRun');
+        this.logger.debug("testsToRun: " + this.testToConvert);
+    }
+
+    protected async additionalConfig(octaneSDKConnection, ws) {
+        this.ciServer = await this.getCiServer(octaneSDKConnection, this.agentJobName === BaseTask.ALM_OCTANE_PIPELINE_START ||
```

Comment:
It seems little messy regarding call getCiServer method on pipeline stat on Base task, we should consider refactor it.